### DISCUSSION TITLE
change default image for publisher in Data Catalog App

### DISCUSSION THREAD

**Discussion by stefan-korn** [GetDKAN/dkan#4114] [Q&A]
First of all, sorry for posting this in the DKAN discussions while it is related to the Data Catalog App. But Data Catalog App does not have discussion, and I did not want to post this as an issue.

So there is a default image that is used for a publisher in data catalog app if no imageURL has been defined in the publishers.json in the Data Catalog App. It is: https://s3.amazonaws.com/dkan-default-content-files/files/group.png

Now if I want to change this image which is defined here as far as i can see: 
https://github.com/GetDKAN/data-catalog-components/blob/main/src/components/Organization/index.jsx#L69

Do I need to change this in the Data Catalog Components or is there an easy/recommend way to change it in the Data Catalog App (I do not consider replacing the the URL in the dist of Data Catalog App a valid approach).

Please bear with me, if this in only a lack of understanding React.

**Comment by janette** [comment]
@stefan-korn you can replace NULL("") with a url to a custom default image here:

-  https://github.com/GetDKAN/data-catalog-app/blob/efee2dce64c32fe949af18ae52d16ac066b4071e/src/templates/dataset/index.jsx#L41
- https://github.com/GetDKAN/data-catalog-app/blob/efee2dce64c32fe949af18ae52d16ac066b4071e/src/templates/dataset/api.jsx#L28

**Comment by stefan-korn** [comment]
@janette : Thanks for you reply.

Okay, it would work changing it for the dataset view changing it in index.jsx like you described, but only if you change the if condition here too:
https://github.com/GetDKAN/data-catalog-app/blob/efee2dce64c32fe949af18ae52d16ac066b4071e/src/templates/dataset/index.jsx#L44
Since the if is now checking for orgDetails[0].imageUrl it will not have any effect. So you would need to change the if to check for orgImage instead.

But still changing it here would not change it on the "publisher" view. I suppose publisher view is taking its list from here:
https://github.com/GetDKAN/data-catalog-app/blob/efee2dce64c32fe949af18ae52d16ac066b4071e/src/templates/publishers/index.jsx
and this template is again going back to data-catalog-components probably.

And also it seems not really DRY having to change it in two places for dataset view and api view.

**Comment by janette** [comment]
If you have images for all publishers then just edit the src/assets/publishers.json in your data-catalog-app. I thought you were asking how to set up a different default image if a publisher did not have one defined under assets.

**Comment by stefan-korn** [comment]
It is about setting up a different default image if a publisher did not have one defined under assets.

**Comment by brdunfield** [comment]
Hello,
It looks like we don't currently offer the ability to override that default from data-catalog-app. Both data-catalog-app and data-catalog-components are evolving boilerplates for starting a new site, and we will work on making it easier to customize in the future. Thank you for your feedback!

In the short term, your best bet would be to copy the relevant components (PublisherList, Organization) from data-catalog-components into your app and make the changes there. The default image is set as a default prop in the Organization component.

**Comment by stefan-korn** [comment]
Okay, thanks,

So I need to copy PublisherList and Organization from data-catalog-components src/components over to data-catalog-app src/components and then I need to change the imports in data-catalog-app like in src/template/publishers/index.jsx and src/templates/dataset/index.jsx to point to the src/components in data-catalog-app instead of the data-catalog-components?

**Reply by brdunfield** [reply]
That sounds correct!
